model/stats/query_test.go

```diff
@@ -0,0 +1,523 @@
+	s.checkTestStats(docs[0], "test1", "task1", "", "", day1, 35, 14, float32(29.834286))
+	s.checkTestStats(docs[1], "test1", "task2", "", "", day1, 4, 4, float32(4))
+
+	// Query with grouping by test
+	// The previous stats should be grouped
+	filter.GroupBy = GroupByTest
+	// Adding a document for test2+task2+v2+d2 that should appear separately in the results
+	s.insertDailyTestStats("p1", "r1", "test2", "task2", "v2", "d2", day1, 5, 5, 5)
+	docs, err = GetTestStats(&filter)
+	require.NoError(err)
+	require.Len(docs, 2)
+	s.checkTestStats(docs[0], "test1", "", "", "", day1, 39, 18, float32(27.184616))
+	s.checkTestStats(docs[1], "test2", "", "", "", day1, 5, 5, float32(5))
+
+	// Test pagination
+	s.insertDailyTestStats("p2", "r1", "test1", "task2", "v2", "d1", day1, 1, 1, 1)
+	s.insertDailyTestStats("p2", "r1", "test2", "task2", "v1", "d1", day1, 1, 1, 1)
+	s.insertDailyTestStats("p2", "r1", "test3", "task2", "v2", "d1", day1, 1, 1, 1)
+	s.insertDailyTestStats("p2", "r1", "test4", "task1", "v1", "d2", day1, 1, 1, 1)
+	s.insertDailyTestStats("p2", "r1", "test5", "task1", "v2", "d1", day1, 1, 1, 1)
+	s.insertDailyTestStats("p2", "r1", "test6", "task1", "v1", "d1", day1, 1, 1, 1)
+
+	filter.Project = "p2"
+	filter.Tests = nil
+	filter.Limit = 2
+	docs, err = GetTestStats(&filter)
+	require.NoError(err)
+	require.Len(docs, 2)
+	s.checkTestStats(docs[0], "test1", "", "", "", day1, 1, 1, float32(1))
+	s.checkTestStats(docs[1], "test2", "", "", "", day1, 1, 1, float32(1))
+
+	filter.StartAt = &StartAt{Date: day1, Test: "test2"}
+	docs, err = GetTestStats(&filter)
+	require.NoError(err)
+	require.Len(docs, 2)
+	s.checkTestStats(docs[0], "test3", "", "", "", day1, 1, 1, float32(1))
+	s.checkTestStats(docs[1], "test4", "", "", "", day1, 1, 1, float32(1))
+
+	// Test pagination with grouping by test+task+variant+distro
+	filter.GroupBy = GroupByDistro
+	filter.StartAt = nil
+	filter.Limit = 2
+	docs, err = GetTestStats(&filter)
+	require.NoError(err)
+	require.Len(docs, 2)
+	// expecting the results ordered by date/variant/task/test/distro
+	s.checkTestStats(docs[0], "test4", "task1", "v1", "d2", day1, 1, 1, float32(1))
+	s.checkTestStats(docs[1], "test6", "task1", "v1", "d1", day1, 1, 1, float32(1))
+
+	filter.StartAt = &StartAt{Date: day1, Test: "test6", Task: "task1", BuildVariant: "v1", Distro: "d1"}
+	docs, err = GetTestStats(&filter)
+	require.NoError(err)
+	require.Len(docs, 2)
+	s.checkTestStats(docs[0], "test2", "task2", "v1", "d1", day1, 1, 1, float32(1))
+	s.checkTestStats(docs[1], "test5", "task1", "v2", "d1", day1, 1, 1, float32(1))
+
+	filter.StartAt = &StartAt{Date: day1, Test: "test5", Task: "task1", BuildVariant: "v2", Distro: "d1"}
+	docs, err = GetTestStats(&filter)
+	require.NoError(err)
+	require.Len(docs, 2)
+	s.checkTestStats(docs[0], "test1", "task2", "v2", "d1", day1, 1, 1, float32(1))
+	s.checkTestStats(docs[1], "test3", "task2", "v2", "d1", day1, 1, 1, float32(1))
+}
+
+func (s *statsQuerySuite) TestGetTaskStats() {
```

Comment:
maybe it makes sense to splt up this test into subtests or additional suites. diagnosing test failures in really long tests can be sort of tedious, and complicated.

Reply:
Done.